rfcs/003-randomness.md

```diff
@@ -0,0 +1,193 @@
+# RFC: Randomness
+
+Author: José Duarte — @jmg-duarte
+Date: 6/12/24
+
+## Abstract
+
+This document covers existing randomness methods in Polkadot and discusses which
+would be appropriate to the Polka Storage project.
+
+There is an implict assumption that the Storage Provider only observes the
+finalized chain
+
+## Introduction
+
+The Polka Storage project requires randomness as a form of creating challenges
+for Storage Providers, with the aim of stopping them from discarding the files
+they are supposed to store, by pre-computing proofs for upcoming challenges.
+Achieving this randomness is not trivial — for example, we cannot allow the
+Storage Provider to generate the randomness alone since they may be dishonest
+and generate values that allow them to fake the storage proofs; furthermore by
+their very nature, blockchains must be somewhat deterministic and as such,
+randomness methods achieved on-chain are not suitable for all use cases. For
+example consider an application that requires multiple rounds of randomness,
+to achieve that you would generally reach out to a pseudo random number
+generator (PRNG), however that PRNG requires a seed to get started, and
+on-chain, everyone can see that seed and as such, predict the outcomes; now
+consider that you changed your code to change the seed every time a new block
+is generated, while the seed is no longer predictable, it is still possible
+for users to read it (it is the latest block hash after all) and predict
+future outcomes while a new block is not available — as such, most methods
+discussed in this RFC are not suitable for cases where multiple random values
```

Comment:
What's the scenario where multiple random values are needed and this approach doesn't work?